pkg/vsphere/disk/disk_manager.go

```diff
@@ -443,8 +443,15 @@ func (m *Manager) devicePathByURI(op trace.Operation, datastoreURI *object.Datas
 }
 
 // AttachAndMount creates and attaches a vmdk as a non-persistent disk, mounts it, and returns the mount path.
-func (m *Manager) AttachAndMount(op trace.Operation, datastoreURI *object.DatastorePath) (string, error) {
-	config := NewNonPersistentDisk(datastoreURI)
+func (m *Manager) AttachAndMount(op trace.Operation, datastoreURI *object.DatastorePath, persistent bool) (string, error) {
+	var config *VirtualDiskConfig
+
+	if !persistent {
+		config = NewNonPersistentDisk(datastoreURI)
```

Comment:
Why is there `NewNonPersistentDisk` and `NewPersistentDisk`, and not just a `persistent` flag that is passed into one function?